File: apps/web/src/types/eventLog.ts
export type EventLogType = "user_behavior" | "domain_event" | "system_event";

export type EventLogSource = "frontend" | "backend" | "script";

export type EventLogCreateRequest = {
  event_name: string;
  event_type: EventLogType;
  source: EventLogSource;
  user_id?: string | null;
  session_id?: string | null;
  entity_type?: string | null;
  entity_id?: string | null;
  properties?: Record<string, unknown>;
};

export type EventLogCreateResponse = {
  event_id: string;
  event_name: string;
  event_type: EventLogType;
  user_id: string | null;
  session_id: string | null;
  entity_type: string | null;
  entity_id: string | null;
  occurred_at: string;
  source: EventLogSource;
  properties: Record<string, unknown>;
  created_at: string;
  message: string;
};

File: apps/api/backend/schemas/event_log.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field


class EventLogCreateRequest(BaseModel):
    event_name: str = Field(..., min_length=1, max_length=100)
    event_type: str = Field(..., min_length=1, max_length=50)
    user_id: UUID | None = None
    session_id: UUID | None = None
    entity_type: str | None = Field(default=None, max_length=50)
    entity_id: UUID | None = None
    source: str = Field(..., min_length=1, max_length=50)
    properties: dict[str, Any] = Field(default_factory=dict)


class EventLogCreateResponse(BaseModel):
    event_id: UUID
    event_name: str
    event_type: str
    user_id: UUID | None = None
    session_id: UUID | None = None
    entity_type: str | None = None
    entity_id: UUID | None = None
    occurred_at: datetime
    source: str
    properties: dict[str, Any]
    created_at: datetime
    message: str